### DISCUSSION TITLE
`useTranslations` no longer working in client components on Next 15

### DISCUSSION THREAD

**Discussion by nekode-dev** [amannn/next-intl#1468] [Q&A]
Hey there,

Since the migration over to app router last year, we took the approach of passing all messages to the client side through the `NextIntlClientProvider` as outlined in the docs [here](https://next-intl-docs.vercel.app/docs/environments/server-client-components#option-4-providing-all-messages). With this approach, we were able to call `useTranslations` directly within any component marked with `'use client'`, as well as all components rendered on the server.

However, since upgrading to Next 15 we are now met with the `”Failed to call useTranslations because the context from NextIntlClientProvider was not found.”` exception outlined in the docs [here](https://next-intl-docs.vercel.app/docs/environments/server-client-components#missing-context). We have tried following the suggestions highlighted in the docs, but we can only really see a way forward by prop drilling all of our translations through to our client components from server components, which seems extremely cumbersome.

I can't find any documentation updates that specify `useTranslations` will no longer work in client components, so is this now intended behaviour or a mistake?

Thanks in advance,

Tom

**Comment by amannn** [comment]
Hey @nekode-dev, that seems odd to me—I didn't have any such issues in my tests with Next.js 15. `useTranslations` is absolutely still supported in Client Components.

Can you by chance share a [reproduction](https://github.com/amannn/next-intl-bug-repro-app-router)?

**Reply by nekode-dev** [reply]
Hey @amannn, thanks for the reply!

Yeah of course, I'll create one now for you in that repo. Hopefully won't be too long! 🙏🏻

**Reply by nekode-dev** [reply]
Hey @amannn, so after checking out your template and getting the reproduction set up, it does indeed work as you say.

This eludes to something being wrong in our repo clearly that we need to dig further into and try and solve. I will get back to you on any updates we come across on this.

Thanks!

**Comment by jbrown** [comment]
@nekode-dev I'm currently having this exact issue with Next v15.2.0 and next-intl v3.26.5, was there every any resolution for this?
